testutils/mongo_client.go

```diff
@@ -9,29 +9,45 @@ import (
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
+const (
+	DefaultStandaloneMongoDBServerURL = "mongodb://localhost:27027"
+	DefaultShardedMongoDBServerURL    = "mongodb://localhost:27037"
+	DefaultReplSetMongoDBServerURL    = "mongodb://localhost:27047"
+)
+
 // MustGetConnectedReplSetClient return mongo.Client instance connected to server started in replicaSet mode.
 func MustGetConnectedReplSetClient(ctx context.Context, t *testing.T) *mongo.Client {
 	opts := options.Client().
-		ApplyURI("mongodb://127.0.0.1:27019/admin").
-		SetReplicaSet("rs0").
+		ApplyURI(DefaultReplSetMongoDBServerURL + "/admin").
+		SetReplicaSet("rs3").
 		SetDirect(true).SetServerSelectionTimeout(time.Second)
 	client, err := mongo.Connect(ctx, opts)
 	if err != nil {
 		t.Fatalf("Couldn't connect to MongoDB instance, reason: %v", err)
 	}
 
+	err = client.Ping(context.Background(), nil)
+	if err != nil {
+		t.Skipf("Test has been skipped, reason: %v", err)
+	}
+
 	return client
 }
 
 // MustGetConnectedMongodClient return mongo.Client instance connected to server started in single mode.
 func MustGetConnectedMongodClient(ctx context.Context, t *testing.T) *mongo.Client {
 	opts := options.Client().
-		ApplyURI("mongodb://127.0.0.1:27017/admin").
+		ApplyURI(DefaultStandaloneMongoDBServerURL + "/admin").
 		SetDirect(true).SetServerSelectionTimeout(time.Second)
 	client, err := mongo.Connect(ctx, opts)
 	if err != nil {
 		t.Fatalf("Couldn't connect to MongoDB instance, reason: %v", err)
 	}
 
+	err = client.Ping(context.Background(), nil)
+	if err != nil {
+		t.Skipf("Test has been skipped, reason: %v", err)
```

Comment:
Not sure it is a good idea to conditionally skip test in function named MustXXX.

Reply:
My POV on this situation is: "Tests should not fail if some part of the testing environment is not ready." 

They should be skipped and should have understandable messages which not confusing the tester. With this change, tests won't confuse developers and testers and it will be more clear to understand that problem not in tests but in the environment. 

I understand your concerns point about function naming. Maybe we may change the function name.

It's here only because actually the MongoDB-driver doesn't connect to the server. It connects to the server when you make a request so it only places when I can check it.